internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -1467,6 +1478,9 @@ func resourceKubernetesClusterCreate(d *pluginsdk.ResourceData, meta interface{}
 			UpgradeChannel: utils.ToPtr(managedclusters.UpgradeChannelNone),
 		}
 	}
+	if v, ok := d.GetOk("node_os_channel_upgrade"); ok {
+		parameters.Properties.AutoUpgradeProfile.NodeOSUpgradeChannel = utils.ToPtr(managedclusters.NodeOSUpgradeChannel(v.(string)))
```

Comment:
We should use the new pointer package for this 
```suggestion
		parameters.Properties.AutoUpgradeProfile.NodeOSUpgradeChannel = pointer.To(managedclusters.NodeOSUpgradeChannel(v.(string)))
```

Reply:
done.